sdkproject/Assets/Mapbox/Examples/LocationProvider/Scripts/MapMatchingExample.cs

```diff
@@ -0,0 +1,79 @@
+﻿namespace Mapbox.Examples
+{
+	using Mapbox.Unity;
+	using Mapbox.Utils;
+	using Mapbox.Unity.Utilities;
+	using Mapbox.Unity.Map;
+	using System.Collections.Generic;
+	using Mapbox.MapMatching;
+	using UnityEngine;
+
+	public class MapMatchingExample : MonoBehaviour
+	{
+		[SerializeField]
+		AbstractMap _map;
+
+		[SerializeField]
+		LineRenderer _lineRenderer;
+
+		[SerializeField]
+		PlotRoute _originalRoute;
+
+		[SerializeField]
+		Profile _profile;
+
+		MapMatcher _mapMatcher;
+
+		void Awake()
+		{
+			_mapMatcher = MapboxAccess.Instance.MapMatcher;
+		}
+
+		[ContextMenu("Test")]
+		public void Match()
+		{
+			var resource = new MapMatchingResource();
+
+			var coordinates = new List<Vector2d>();
+			foreach (var position in _originalRoute.Positions)
+			{
+				var coord = position.GetGeoPosition(_map.CenterMercator, _map.WorldRelativeScale);
+				coordinates.Add(coord);
+				Debug.Log("MapMatchingExample: " + coord);
+			}
+
+			resource.Coordinates = coordinates.ToArray();
+			resource.Profile = _profile;
+			_mapMatcher.Match(resource, HandleMapMatchResponse);
+		}
+
+		void HandleMapMatchResponse(MapMatching.MapMatchingResponse response)
+		{
+			if (response.HasMatchingError)
+			{
+				Debug.LogError("MapMatchingExample: " + response.MatchingError);
+				return;
+			}
+
+			_lineRenderer.positionCount = 0;
+			for (int i = 0, responseTracepointsLength = response.Tracepoints.Length; i < responseTracepointsLength; i++)
+			{
+				var point = response.Tracepoints[i];
+
+				// FIXME: why/how can a point be null?
```

Comment:
@BergWerkGIS Any idea why I have to check if these points are null? How is that possible?

Reply:
https://github.com/mapbox/mapbox-unity-sdk/blob/map-matching/sdkproject/Assets/Mapbox/Core/mapbox-sdk-cs/MapMatching/MapMatchingResource.cs#L127-L128

https://github.com/mapbox/mapbox-unity-sdk/blob/map-matching/sdkproject/Assets/Mapbox/Core/mapbox-sdk-cs/Tests/UnitTests/Editor/MapMatcherTest.cs#L467

Reply:
Also:
> If a trace point is omitted by map matching because it is an outlier, the entry will be null.

https://www.mapbox.com/api-documentation/#match-response-object

Reply:
Thanks for the clarification. I read that, but assumed those entries would have simply been excluded from the server (and I wasn't using `Tidy`). Strange.

Reply:
To be able to evaluate results with regards to input data all arrays (coordinates, tracepoints, radiuses, timestamps, ...) of requests and responses have to be of same length always.